### ISSUE TITLE
Feature request: tiling backgrounds

### ISSUE THREAD

**Issue by triptych** [replit/kaboom#79]
making a space game and I want a background that tiles automatically. Right now Ill have to manually create sprites that line up correctly. Would be great if there was a sprite mode that auto tiles the image provided based on the dimensions.
Like background images in HTML.

**Comment by slmjkdbtl** [comment]
Besides drawing multiple images, there's another way: If your image's dimension is power of 2 and you specify the `quad` to be bigger than `(0, 0, 1, 1)`, it'll use `GL_REPEAT` wrapping, so a
```js
add([
    sprite("tile", {
        quad: quad(0, 0, 2, 2),
    }),
]);
```
but this only works for power of 2 images. I'll think about if there's a good interface for this. Maybe something like
```
sprite("tile", {
    width: 240,
    height: 240,
    mode: "tile", // or "stretch", for controlling the behavior with size is explicitly set
});
```

**Comment by slmjkdbtl** [comment]
Implemented in 13936ee, and example https://kaboomjs.com/examples#tiled
```js
add([
	sprite("grass", {
		width: width(),
		height: height(),
		tiled: true,
	}),
]);
```

**Comment by triptych** [comment]
@slmjkdbtl The example doesn't work :
<img width="885" alt="vivaldi_orKNWXMEsI" src="https://user-images.githubusercontent.com/8691/125017245-94d0f280-e027-11eb-91e9-2e0b81389f83.png">
